test/test_torch.py

```diff
@@ -6397,80 +6398,210 @@ def test_logical_and(self, device):
-            res2[i] = max(min(res2[i], max_val), min_val)
-        self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+        m1_res = torch.clamp(m1, min_val, max_val)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(m1_res))
 
-        out = test_tens.clone()
-        torch.clamp(test_tens, min=min_val, max=max_val, out=out)
-        self.assertEqual(torch.isnan(out), torch.isnan(res1))
+        out = torch.empty_like(m1)
+        torch.clamp(m1, min_val, max_val, out=out)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(out))
 
-        res1 = torch.clamp(test_tens, min=min_val)
-        res2 = test_tens.clone()
-        for i in iter_indices(res2):
-            res2[i] = max(res2[i], min_val)
-        self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+        m1.clamp_(min_val, max_val)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(m1))
 
-        torch.clamp(test_tens, min=min_val, out=out)
-        self.assertEqual(torch.isnan(out), torch.isnan(res1))
+    def test_clamp_min_propagates_nans(self, device):
+        min_val = -1
+        m1, m1_base = self.generate_clamp_baseline(device, min_vals=min_val, with_nans=True)
 
-        res1 = torch.clamp(test_tens, max=max_val)
-        res2 = test_tens.clone()
-        for i in iter_indices(res2):
-            res2[i] = min(res2[i], max_val)
-        self.assertEqual(torch.isnan(res1), torch.isnan(res2))
+        m1_res = torch.clamp(m1, min=min_val)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(m1_res))
 
-        torch.clamp(test_tens, max=max_val, out=out)
-        self.assertEqual(torch.isnan(out), torch.isnan(res1))
+        out = torch.empty_like(m1)
+        torch.clamp(m1, min=min_val, out=out)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(out))
 
+    def test_clamp_max_propagates_nans(self, device):
+        max_val = 1
+        m1, m1_base = self.generate_clamp_baseline(device, max_vals=max_val, with_nans=True)
+
+        m1_res = torch.clamp(m1, max=max_val)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(m1_res))
+
+        out = torch.empty_like(m1)
+        torch.clamp(m1, max=max_val, out=out)
+        self.assertEqual(torch.isnan(m1_base), torch.isnan(out))
+
+    def test_clamp_raises_arg_errors(self, device):
+        m1 = torch.randn(100, dtype=torch.float, device=device)
         error_msg = 'At least one of \'min\' or \'max\' must not be None'
         with self.assertRaisesRegex(RuntimeError, error_msg):
             m1.clamp()
         with self.assertRaisesRegex(RuntimeError, error_msg):
             m1.clamp_()
+        with self.assertRaisesRegex(RuntimeError, error_msg):
+            torch.clamp(m1)
+
+    def test_clamp_with_tensors(self, device):
+        min_vals = torch.linspace(-1, 0, 100, device=device)  # [-1..0]
+        max_vals = torch.linspace(1, 0, 100, device=device)  # [1..0]
+        m1, m1_base = self.generate_clamp_baseline(device, min_vals, max_vals)
```

Comment:
This is where you might want to follow the "expected," from NumPy, and "actual," from PyTorch, pattern found in other parts of the test suite. You'll still need a helper to your tensors vals but I think it would be simpler and (again) ensure our clamp is like NumPy's.